File: Array_Functions/Prefix_Sum/prefix_sum.py
def generate_prefix_sum(iterable, start, end):
    if start < 1 or end > len(iterable) or start > end:
        print("Invalid range. Please ensure the range is within bounds and start <= end.")
        return []

    prefix_sum = [0]
    current_sum = 0
    for i in range(start - 1, end):  # Adjusting for 1-based indexing
        current_sum += iterable[i]
        prefix_sum.append(current_sum)
    
    return prefix_sum

if __name__ == "__main__":
    # Example with a list 
    a = [1, 2, 3, 4, 5]
    start_index = 2
    end_index = 4
    
    prefix_sum_a = generate_prefix_sum(a, start_index, end_index)
    if prefix_sum_a:  # Check if the range is valid and prefix sum is not empty
        print("Prefix Sum for range (array):", " ".join(map(str, prefix_sum_a)))
    
  

File: Array_Functions/Prefix_Sum/prefix_sum.java
package CodeStock.Array_Functions.Prefix_Sum;

import java.util.*;

public class prefix_sum {

    public static List<Integer> generatePrefixSum(int[] array, int start, int end) {
        if (start < 1 || end > array.length || start > end) {
            System.out.println("Invalid range for array. Please check the start and end indices.");
            return Collections.emptyList();
        }

        List<Integer> prefixSum = new ArrayList<>();
        prefixSum.add(0);

        int currentSum = 0;

        for (int i = start; i <= end; i++) {
            currentSum += array[i - 1];
            prefixSum.add(currentSum);
        }

        return prefixSum;
    }

    public static List<Integer> generatePrefixSum(List<Integer> list, int start, int end) {
        if (start < 1 || end > list.size() || start > end) {
            System.out.println("Invalid range for list. Please check the start and end indices.");
            return Collections.emptyList();
        }

        List<Integer> prefixSum = new ArrayList<>();
        prefixSum.add(0);

        int currentSum = 0;

        for (int i = start; i <= end; i++) {
            currentSum += list.get(i - 1);
            prefixSum.add(currentSum);
        }

        return prefixSum;
    }

    public static void main(String[] args) {
        //example as array
        int[] a = {1, 2, 3, 4, 5};
        int startIndex = 1;
        int endIndex = 4;

        List<Integer> prefixSumArray = generatePrefixSum(a, startIndex, endIndex);

        if (!prefixSumArray.isEmpty()) {
            System.out.print("Prefix Sum for array subrange: ");
            for (int sum : prefixSumArray) {
                System.out.print(sum + " ");
            }
            System.out.println();
        }
        //example as list
        List<Integer> list = Arrays.asList(1, 2, 3, 4, 5);
        startIndex = 1;
        endIndex = 4;

        List<Integer> prefixSumList = generatePrefixSum(list, startIndex, endIndex);

        if (!prefixSumList.isEmpty()) {
            System.out.print("Prefix Sum for list subrange: ");
            for (int sum : prefixSumList) {
                System.out.print(sum + " ");
            }
            System.out.println();
        }
    }
}
